hack/tools/release/notes/github.go

```diff
@@ -0,0 +1,205 @@
+//go:build tools
```

Comment:
I decided to keep using `gh` for now since that was the previous dependency so everyone who was using it already had it setup. It already handles authentication so I didn't have to document/add config how to authenticate against github.

I think switching this to a github library is a good improvement, it should simplify this code. However, I would prefer to have it as an improvement on top of this, to minimize disruption by this first refactor.

Reply:
Sounds good.